src/main/java/com/hermesworld/ais/galapagos/topics/service/impl/ValidatingTopicServiceImpl.java

```diff
@@ -159,8 +166,20 @@ public boolean canDeleteTopic(String environmentId, String topicName) {
         return checkOnNonStaging(environmentId, action, Void.class);
     }
 
+    private Optional<CompletableFuture<Void>> checkOnNonStaging(String environmentId, String action, String topicName) {
```

Comment:
This method does not (only) do what the name suggests. The check for subscribers should not be in such a generic method.